target/scripts/startup/setup.d/log.sh

```diff
@@ -13,31 +13,25 @@ function _setup_logs_general() {
 function _setup_logrotate() {
   _log 'debug' 'Setting up logrotate'
 
-  LOGROTATE='/var/log/mail/mail.log\n{\n  compress\n  copytruncate\n  delaycompress\n'
-
-  case "${LOGROTATE_INTERVAL}" in
-    ( 'daily' )
-      _log 'trace' 'Setting postfix logrotate interval to daily'
-      LOGROTATE="${LOGROTATE}  rotate 4\n  daily\n"
-      ;;
-
-    ( 'weekly' )
-      _log 'trace' 'Setting postfix logrotate interval to weekly'
-      LOGROTATE="${LOGROTATE}  rotate 4\n  weekly\n"
-      ;;
-
-    ( 'monthly' )
-      _log 'trace' 'Setting postfix logrotate interval to monthly'
-      LOGROTATE="${LOGROTATE}  rotate 4\n  monthly\n"
-      ;;
-
-    ( * )
-      _log 'warn' 'LOGROTATE_INTERVAL not found in _setup_logrotate'
-      ;;
-
-  esac
-
-  echo -e "${LOGROTATE}}" >/etc/logrotate.d/maillog
+  if [[ ${LOGROTATE_INTERVAL} =~ ^(dai|week|month)ly$ ]]; then
+    _log 'trace' "Logrotate interval set to ${LOGROTATE_INTERVAL}"
+  else
+    _log 'warn' "Value '${LOGROTATE_INTERVAL}' for LOGROTATE_INTERVAL is invalid - falling back to default"
+    export LOGROTATE_INTERVAL='weekly'
+    # shellcheck disable=SC2034
+    VARS[LOGROTATE_INTERVAL]='weekly'
```

Comment:
The default is already set by this point?:

https://github.com/docker-mailserver/docker-mailserver/blob/82c38f242648290ab109de8e3994ea7564120deb/target/scripts/startup/variables-stack.sh#L138

https://github.com/docker-mailserver/docker-mailserver/blob/82c38f242648290ab109de8e3994ea7564120deb/target/scripts/start-mailserver.sh#L154

Recommending a [panic](https://github.com/docker-mailserver/docker-mailserver/blob/82c38f242648290ab109de8e3994ea7564120deb/target/scripts/helpers/error.sh#L21) for misconfiguration here during startup. We should clearly communicate to the user that they've explicitly set an incompatible ENV value for the setting and bail, just like we do with TLS certs during startup:

https://github.com/docker-mailserver/docker-mailserver/blob/82c38f242648290ab109de8e3994ea7564120deb/target/scripts/helpers/ssl.sh#L368

https://github.com/docker-mailserver/docker-mailserver/blob/82c38f242648290ab109de8e3994ea7564120deb/target/scripts/helpers/error.sh#L41-L47

---

**Mandatory:**

```suggestion
    _dms_panic__invalid_value 'LOGROTATE_INTERVAL' 'Setup -> Logrotate'
```